desc/objectives/utils.py

```diff
@@ -150,22 +139,21 @@ def recover(x_reduced):
         return jnp.atleast_1d(jnp.squeeze(xp + dx))
 
     # check that all constraints are actually satisfiable
-    xp_ = objective.unpack_state(xp, False)
-    for con in constraints:
-        xpi = [xp_[i] for i, t in enumerate(objective.things) if t in con.things]
+    params = objective.unpack_state(xp, False)
+    for con in constraint.objectives:
+        xpi = [params[i] for i, t in enumerate(objective.things) if t in con.things]
         y1 = con.compute_unscaled(*xpi)
         y2 = con.target
         y1, y2 = np.broadcast_arrays(y1, y2)
         np.testing.assert_allclose(
             y1,
             y2,
-            atol=2e-14,
-            rtol=5e-14,
-            err_msg="Incompatible constraints detected, cannot satisfy "
-            + f"constraint {con}",
+            atol=3e-14,
```

Comment:
I think we should keep an rtol here in case of units issues.

Also I wonder if this might be better as a warning than an error? I've run into cases where due to units or scale issues it fails with an error of 1e-12 or 1e-10 or something, not a showstopper but something the user should at least be aware of.

Reply:
What do you mean by unit issues? The only reason this won't be exactly satisfied is from numerical issues with inverting the `A` matrix, right? So I guess if different constraints have different orders of magnitude it could get messy.

Reply:
I can add the rtol back in. I'm on the fence between error vs warning

Reply:
yeah "unit issues" being constraints with different units that are different orders of magnitude, like pressure/current/temperature etc. I've also run into the same issue if I just re-scale an equilibrium to something really big

Reply:
I think I am in favor of moving to a warning but let's do that in another PR